mlflow/tensorflow/__init__.py

```diff
@@ -740,6 +740,16 @@ def autolog(
 
     atexit.register(_flush_queue)
 
+    if log_model_signatures:
+        warnings.warn(
+            "When 'log_model_signatures' is set to True, "
+            "it can result in output types that differ from the input type"
+            " for predict using PyFunc implementation. "
+            "For example, if pandas DF is used for prediction input, "
+            "the output could be ndarray or dict of (str -> ndarray) based "
+            "on what the signature of the model was inferred."
```

Comment:
Can we move this warning to `mlflow.tensorflow.log_model()`? This applies beyond autologging. I think the following text may be clearer:

```
            " The pyfunc inference behavior of TensorFlow models logged with signatures differs"
            " from the behavior of TensorFlow models logged without signatures. Specifically, when a"
            " signature is present, passing a Pandas DataFrame as input to the pyfunc `predict()` API"
            " produces an `ndarray` (for single-output models) or a dictionary of `str -> ndarray`:"
            " (for multi-output models). In contrast, when a signature is *not* present, `predict()` produces"
            " a Pandas DataFrame output in response to a Pandas DataFrame input."
```
            
            
Can we also use `_logger.warning()` rather than `warnings.warn()`?

Reply:
This applies to `keras` models as well, right? In that case, we should also add the warning to `mlflow.keras.log_model()`